### PR TITLE
Added Support for IOS-XR 

### PR DESCRIPTION
**Pull request by ajitdani**
Create this as separate branch as had forgot to include unit test cases

### CODE REVIEW

**Comment by ajitdani** [comment]
Hello, Can this be merged into master ?

File: test/unit/operations/third_party/iosxr/test_rpc.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from ncclient.operations import RaiseMode
+from xml.etree import ElementTree
+
+
+class TestRPC(unittest.TestCase):
+
+    @patch('ncclient.transport.SSHSession')
+    @patch('ncclient.operations.third_party.iosxr.rpc.RPC._request')
+    def test_execute_rpc(self, mock_request, mock_session):
+        device_handler = manager.make_device_handler({'name': 'junos'})
```
Comment (einarnn) [inline test/unit/operations/third_party/iosxr/test_rpc.py]: Why is a test case, apparently for the IOS XR device handler, referring to the junos device handler?

File: ncclient/operations/third_party/iosxr/rpc.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from ncclient.xml_ import *
+
+from ncclient.operations.rpc import RPC
+from ncclient.operations.rpc import RPCReply
+from ncclient.operations.rpc import RPCError
+from ncclient import NCClientError
+
+class ExecuteRpc(RPC):
```
Comment (einarnn) [inline ncclient/operations/third_party/iosxr/rpc.py]: Can you please explain what this class is doing and why? I think I can see what you are trying to achieve, but please be explicit!

File: ncclient/operations/third_party/iosxr/rpc.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    def request(self, rpc, filter_xml=None):
+        if isinstance(rpc, str):
```
Comment (einarnn) [inline ncclient/operations/third_party/iosxr/rpc.py]: What happens if I pass something apart from a string? Should the string be checked for inclusion of bad characters? E.g. a string looking like "<rpc-name/>" will be accepted. What happens if an lxml.etree object is passed in? Etc.

Need better error checking at minimum.

File: ncclient/operations/third_party/iosxr/rpc.py (hunk 2 above)
Comment (ajitdani) [reply ncclient/operations/third_party/iosxr/rpc.py]: Currently the default device handler ends up using Junos Device handler, due to which common RPCs like commit ends up failing on Cisco IOS-XR devices as example Commit rpc in juniper is <commit-configuration/> however in Cisco it is <commit/>, currently to workaround this, I have just mirrored the juniper ExecuteRPC class and added it in Cisco IOS-XR handler to allow us to proceed with the testing.

File: test/unit/operations/third_party/iosxr/test_rpc.py (hunk 1 above)
Comment (ajitdani) [reply test/unit/operations/third_party/iosxr/test_rpc.py]: nice catch, I have fixed this in latest commit

File: ncclient/operations/third_party/iosxr/rpc.py (hunk 2 above)
Comment (einarnn) [reply ncclient/operations/third_party/iosxr/rpc.py]: I have been using the default device handler successfully for RPCs such as commit with IOS XR for a long time, so I’m a little confused. What problems have you been experiencing?

Comment (ajitdani) [reply ncclient/operations/third_party/iosxr/rpc.py:8]: When I use default handler it doesnt generate the right commit rpc 
example below 
<?xml version="1.0" encoding="UTF-8"?><nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:18792888-9b0a-4847-a098-1be601c479e1"><commit-configuration/></nc:rpc>

Comment (ajitdani) [reply ncclient/operations/third_party/iosxr/rpc.py:8]: Just to add - custom RPC does work with default handler

Comment (einarnn) [reply ncclient/operations/third_party/iosxr/rpc.py:8]: > When I use default handler it doesnt generate the right commit rpc
> example below:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:18792888-9b0a-4847-a098-1be601c479e1">
>   <commit-configuration/>
> </nc:rpc>
> ```

This example is from the JunOS handler, AFAIK. As I said, I've been using the default device handler, and others the IOS XR handler, for a number of years, including commit operations.

Comment (einarnn) [reply ncclient/operations/third_party/iosxr/rpc.py:8]: Can you please provide a small test harness that demonstrates the problem you have encountered? There should be no code path that allows the JunOS device handler to provide operations for the default device handler.

**Comment by einarnn** [comment]
Waited too long for requested changes/feedback. Please reopen with updates if you still think this is required. Note that there was a recent commit to make vendor device additional RPCs **not**global. This may resolve your issue.

**Pull request closed without merging**